packages/ui/AddLogoButton/index.tsx

```diff
@@ -0,0 +1,131 @@
+"use client";
+
+import { Pencil } from "lucide-react";
+import Image from "next/image";
+import { useState } from "react";
+import toast from "react-hot-toast";
+
+import { TMembershipRole } from "@formbricks/types/memberships";
+import { TProduct } from "@formbricks/types/product";
+import { TSurvey } from "@formbricks/types/surveys";
+
+import { Input } from "../Input";
+import { Label } from "../Label";
+import { LogoSettingModal } from "../LogoSettingModal";
+import { uploadLogo } from "./lib/uploadLogo";
+
+interface Props {
+  environmentId: string;
+  product: TProduct;
+  type?: string;
+  membershipRole?: TMembershipRole;
```

Comment:
Won't this always be present?

Reply:
type can be made permanent, but membershipRole to be made  permanent ,then it should be permanent from the PreviewSurvey file.and preview survey file is being used at other places and from those these are not being provided,thats why optional.